monitor_check.py

```diff
@@ -201,6 +208,55 @@ async def guess_reason(session, url, http_semaphore):
             }
     return None
 
+async def guess_missing_dependency(session, package, build, http_semaphore):
+    url = builderlive_link(package, build)
+    try:
+        content = await fetch(session, url, http_semaphore)
+    except aiohttp.client_exceptions.ClientPayloadError:
+        logger.debug('broken content %s', url)
+        return False
+    match = re.search(r"Problem: package (.*?) requires", content)
+    if match:
+        pkg = pkgname(match.group(1))
+        if pkg not in missing_dependencies:
+            missing_dependencies[pkg] = []
+            missing_dependencies[pkg].append(package)
+        else:
+            if package not in missing_dependencies[pkg]:
+                missing_dependencies[pkg].append(package)
+    else:
+        missing_dependencies['match_failed'].append(package)
+
+def print_dependency_tree():
+    root = Node("root")
+
+    for k, v in missing_dependencies.items():
+        # insert keys as root children if they are not alredy in tree
+        if len(findall_by_attr(root,k)) == 0:
+            Node(k, parent=root)
+
+        for child in v:
+            # get parent
+            existing_parent = findall_by_attr(root, k)
+            # add child
+            Node(child, parent=existing_parent[0])
+            # get all childs with same name
+            existing_child = findall_by_attr(root, child)
+            # there never should be more than 2 of them
+            if len(existing_child) > 1:
+                # assign new parent to the older child (one with whole dependency tree)
+                existing_child[0].parent=existing_parent[0]
+                # remove the other duplicate child
+                existing_child[1].parent=None
+
+    for pre, _, node in RenderTree(root):
+        print("%s%s" % (pre, node.name))
```

Comment:
Let's stick to one way to format strings in this file?

```suggestion
        print(f"{pre}{node.name}")
```